tests/vmi_cloudinit_test.go

```diff
@@ -174,7 +174,9 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 		Context("with cloudInitConfigDrive userDataBase64 source", func() {
 			It("[test_id:3178]should have cloud-init data", func() {
 				userData := fmt.Sprintf("#!/bin/sh\n\ntouch /%s\n", expectedUserDataFile)
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndConfigDriveUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), userData)
+				vmi := libvmi.NewCirros(
+					libvmi.WithCloudInitConfigDriveUserData(userData),
```

Comment:
This test is about the base64 encoded user data in cloudInitConfigDrive:
```suggestion
					libvmi.WithCloudInitConfigDriveEncodedUserData(userData),
```
This option function does not exist and should be added.

Reply:
The test is not about the encoded base64, but the non-encoded one.
Also, the test previously was already using non-encoded data:
```
AddCloudInitConfigDriveData(vmi, "disk1", userData, "", false) //false means non-encoded
```

Reply:
Either the test or the title is misleading.

Reply:
Got you! Thanks! Looking at the history of the test, it looks like it was created with no encoding, so this makes me think that it is the test description that is misleading.

Reply:
My guess is the test author copy-pasted the CloudInitNoCloud tests but forgot to remove the 'base64' part from description ([PR](https://github.com/kubevirt/kubevirt/pull/2120)).

In terms of coverage we are good because there are [unit tests](https://github.com/kubevirt/kubevirt/blob/main/pkg/cloud-init/cloud-init_test.go#L464) that covers both options (use-data and base64-user-data).

I think we should rename the test description to not mention base64 as history shows the test verifies regular user-data and we have base64 logic covered in unit-tests.
@orelmisan @fossedihelm  WDYT?

Reply:
AFAIU, `[test_id:3180] with cloudInitConfigDrive userData source"` is testing it.
So it makes sense to me to test the base64 logic.

Reply:
I'm ok with both of the decisions:
- change the test description
- test w/ the encoded data, adding a new `Option` (in a separate commit please).

However, the test duplication @orelmisan discovered, deserve a punctual attention and maybe a revision of the entire cloudinit e2e test files.

Reply:
Done, test is fixed to verify encoded user data.

On a followup PR we can revisit this whole test suite and consolidate/remove tests.